### PR TITLE
openai[patch]: type reasoning_effort

### PR DESCRIPTION
**Pull request by baskaryan**

### CODE REVIEW

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 1]
```diff
@@ -454,6 +454,16 @@ class BaseChatOpenAI(BaseChatModel):
     """Total probability mass of tokens to consider at each step."""
     max_tokens: Optional[int] = Field(default=None)
     """Maximum number of tokens to generate."""
+    reasoning_effort: Optional[str] = None
```
Comment (ccurme) [inline libs/partners/openai/langchain_openai/chat_models/base.py:457]: should we type as literal?

**Review by ccurme — APPROVED** [review]
I'm ignorant on this but what is state of o1 support in AzureChatOpenAI?

File: libs/partners/openai/langchain_openai/chat_models/base.py (hunk 1 above)
Comment (baskaryan) [reply libs/partners/openai/langchain_openai/chat_models/base.py:457]: doesn't seem super high value? esp if we care about being forwards compatible

**Comment by baskaryan** [comment]
> I'm ignorant on this but what is state of o1 support in AzureChatOpenAI?

not sure, but nothing gets passed unless explicitly specified

**Pull request merged**